src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int NAME_POSITION = 0;
+    private static final int NUMBER_POSITION = 1;
+
+    private static final int STARTING_NUMBER = 0;
+    private int suplyTotal = 0;
+    private int buyTotal = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        suplyTotal = STARTING_NUMBER;
+        buyTotal = STARTING_NUMBER;
+
+        readFile(fromFileName);
+        createReport(toFileName);
+    }
+
+    private void readFile(String fileName) {
+        File readingFile = new File(fileName);
+        try (BufferedReader reader = new BufferedReader(new FileReader(readingFile))) {
+            String value = reader.readLine();
+            while (value != null) {
+                String[] divider = value.split(",");
+                if (divider[NAME_POSITION].equals("supply")) {
+                    suplyTotal += Integer.parseInt(divider[NUMBER_POSITION]);
+                } else if (divider[NAME_POSITION].equals("buy")) {
+                    buyTotal += Integer.parseInt(divider[NUMBER_POSITION]);
+                }
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't read the file", e);
+        }
+    }
+
+    private void createReport(String fileName) {
```

Comment:
This method is not only creating report, it's also creating file and writes report into this file. Please in getStatistic create three separate methods, readFile(), createReport(), writeFile()